tests/devices/test_gorilla_fan_v2.py

```diff
@@ -0,0 +1,71 @@
+from homeassistant.components.fan import FanEntityFeature
+
+from ..const import FAN_PAYLOAD
+from ..helpers import assert_device_properties_set
+from .base_device_tests import TuyaDeviceTestCase
+
+SWITCH_DPS = "1"
+PRESET_DPS = "2"
+FANMODE_DPS = "3"
+
+class TestGorillaFanV2(TuyaDeviceTestCase):
+    __test__ = True
+
+    def setUp(self):
+        self.setUpForConfig("gorilla_fan_v2.yaml", FAN_PAYLOAD)
+        self.subject = self.entities.get("fan")
+
+    def test_supported_features(self):
+        self.assertEqual(
+            self.subject.supported_features,
+            (
+                FanEntityFeature.PRESET_MODE
+                | FanEntityFeature.SET_SPEED
+            ),
+        )
+
+    def test_preset_mode(self):
+        self.dps[PRESET_DPS] = "normal"
+        self.assertEqual(self.subject.preset_mode, "Normal")
+
+        self.dps[PRESET_DPS] = "boost"
+        self.assertEqual(self.subject.preset_mode, "Boost")
+
+    def test_preset_modes(self):
+        self.assertCountEqual(self.subject.preset_modes, ["normal", "boost"])
+
+    async def test_set_preset_mode_to_normal(self):
+        async with assert_device_properties_set(
```

Comment:
The error here is because of the mapping on preset_mode.  But better would be to use lower case, with translation_key as above, and use "strong" instead of "boost" since that already has translations.

Reply:
removed testcases as they are unnecessary